opensearch-observability/src/main/kotlin/org/opensearch/observability/index/ObservabilityIndex.kt

```diff
@@ -108,6 +113,28 @@ internal object ObservabilityIndex {
         }
     }
 
+    /**
+     * Check if the index mappings have changed and if they have, update them
+     */
+    private fun updateMappings() {
```

Comment:
Why do we need to update mappings? I might be missing something here.

Reply:
Since users will already have their indices with the old mapping that doesn't include the Application object type, we need to update their index to reflect the new mapping that does include the Application object type.